src/app/search/search-config.ts

```diff
@@ -67,6 +67,16 @@ export function buildSearchConfig(
   const allApplicableFilters: FilterDefinition[] = [];
   for (const filter of allFilters) {
     if (!filter.destinyVersion || filter.destinyVersion === destinyVersion) {
+      // augment filter's suggestionKeywords with dynamic additions
+      if (filter.suggestionKeywordGenerator) {
+        const suggestionKeywords = new Set(filter.suggestionKeywords ?? []);
+        for (const keyword of filter.suggestionKeywordGenerator(suggestionsContext) ?? []) {
+          suggestionKeywords.add(keyword);
+        }
+        if (suggestionKeywords.size) {
+          filter.suggestionKeywords = Array.from(suggestionKeywords);
+        }
+      }
```

Comment:
If I'm reading this right, this mutates the static filter definiton objects by saving the dynamically generated `suggestionKeywords` back in L77, and if this runs again with a different context, reads the outdated keywords next run in L72. I think filter def object should be cloned first if we're going with this approach.

Reply:
it does mutate them.
i'm not really sure of the value in maintaining the originals if that's what you intend with cloning,
and it doesn't seem susceptible to causing any particular issue but that's definitely why i wanted your eyes on it

Reply:
my problem is that query validator directly checks the filter def's suggestion keyword array to decide if the search is acceptable, and though i have always wanted a post-processing, normalized version of filter defs with fewer nullable props, it seems like a hassle to set that up just for this

Reply:
oh ok i see the logic problem here